### ISSUE TITLE
Using input labels incorrectly

### ISSUE THREAD

**Issue by manfromjupyter** [nbgallery/nbgallery#771]
On all forms, we only use `<label>`  on radio buttons but we should be using them for all inputs and textareas too.

Reason it has passed all accessibility checks is our placeholder text usually mirrors the label. For all forms the `<span>` label should be replaced with a `<label>`. This is more efficient than adding aria-labels also.

Example of what it is currently:
```html
<span class="input-group-addon input-min-width">Change Log</span>
<textarea class="form-control auto-expand" id="stageSummary" maxlength="250" minlength="3" name="summary" placeholder="Enter a short description of what was changed" required=""></textarea>
```

Change to:
```html
<label class="input-group-addon input-min-width">Change Log</label>
<textarea class="form-control auto-expand" id="stageSummary" maxlength="250" minlength="3" name="summary" placeholder="Enter a short description of what was changed" required=""></textarea>
```
